tss-esapi/src/utils/mod.rs

```diff
@@ -1262,27 +1260,23 @@ impl PcrData {
                     return Err(Error::local_error(WrapperErrorKind::InconsistentParams));
                 }
             }
-            // Add the parsed pcr bank for the parsed hashing algorithm.
-            if parsed_pcr_data
-                .data
-                .insert(parsed_hash_algorithm, parsed_pcr_bank)
-                .is_some()
-            {
-                error!("Error trying to insert data into a PcrBank where data have already been inserted");
-                return Err(Error::local_error(WrapperErrorKind::InconsistentParams));
-            }
+            data.push((parsed_hash_algorithm, parsed_pcr_bank));
```

Comment:
Is it possible this loses track of some invalid cases that the previous version was handling, i.e. having multiple identical hash algorithms?

Reply:
No, because this just adds the additional instance, and then re-builds it.
It's totally possible to have `data: [(sha1, _), (sha256, _), (sha1, _)]`, and this will still allow that.